src/display_size.rs

```diff
@@ -0,0 +1,72 @@
+use crate::common::*;
+
+pub(crate) trait DisplaySize {
+  fn display_size(self) -> Wrapper;
+}
+
+impl DisplaySize for u64 {
+  fn display_size(self) -> Wrapper {
+    Wrapper(self)
+  }
+}
+
+pub(crate) struct Wrapper(u64);
+
+impl Display for Wrapper {
+  fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+    const SUFFIXES: &[&str] = &["B", "KiB", "MiB", "GiB", "TiB", "PiB", "EiB"];
+
+    let value = self.0;
+
+    let (power, suffix) = SUFFIXES
+      .iter()
+      .enumerate()
+      .map(|(i, suffix)| (2u64.pow(i as u32 * 10), suffix))
+      .filter(|(power, _)| value >= power - 1)
+      .last()
+      .unwrap();
+
+    if self.0 % power == 0 {
+      write!(f, "{} {}", self.0 / power, suffix)?;
+    } else {
+      write!(f, "{:.1} {}", self.0 as f64 / power as f64, suffix)?;
+    }
```

Comment:
I would've though it made more sense that
1. sizes in bytes (`B`) always get rendered without decimal points
2. sizes in units above bytes always get rendered with exactly one digit after the decimal point.

That seems more consistent to me. TIOLI.

Reply:
I think you're probably right. I think it's nice that exact figures don't have a decimal point, but people probably won't understand that, and even if they do, it probably isn't worth the visual inconsistency between exact and inexact values.